src/Maestro/SubscriptionActorService/MaestroBarClient.cs

```diff
@@ -0,0 +1,114 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.DotNet.DarcLib;
+using Microsoft.DotNet.Maestro.Client.Models;
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace SubscriptionActorService
+{
+    /// <summary>
+    ///     A bar client interface for use by DarcLib which talks directly
+    ///     to the database for diamond dependency resolution.  Only a few features are required.
+    /// </summary>
+    internal class MaestroBarClient : IBarClient
```

Comment:
Uh... nothing in this file is implemented. Is that expected? Why are we adding it?

Reply:
It will be once i add the coherency support into Maestro itself.  Working on that now.

Reply:
Why add it before we need it?

---
In reply to: [261712294](https://github.com/dotnet/arcade-services/pull/170#discussion_r261712294) [](ancestors = 261712294)

Reply:
Just for completeness.  Habits pulled over from compiler development where we basically NYI/assert as much possible to make it very clear what the green code paths are.